src/Microsoft.DncEng.SecretManager/Readme.md

```diff
@@ -94,17 +88,44 @@ Each service has at least one settings file. These are expected to be named `set
 }
 ```
 
+## Functionality in the Builds (FR Look Here)
+The Secret Manager is set up to run 3 separate parts in our builds: usage validation during the testing phase, vault validation before deployment, and secret rotation on a weekly cadence.
+
+### Usage Validation
+This step runs the `validate-all` command. This command finds all `settings.json` and `settings.<environment>.json` files and validates those files with the manifests in the repo. This step doesn't talk to azure at all, it simply validates that all `[vault()]` references in the settings files have matching specifications in the appropriate manifest. Failures in this step require either adding missing definitions to the manifest, or removing the offending references in the settings file.
+
+### Pre-Deployment Validation
+This step runs the `synchronize --verify-only` command before the "approval" stage for the deployments. This command compares each manifest with the contents of the corresponding vault, and produces errors if secrets in the vault are missing or out-of-date. Failures in this step can be fixed by running the weekly build again, or by manually running the `synchronize` command.
+
+### Weekly Rotation
+This step runs the `synchronize` command. This compares every secret specified in the manifest with the corresponding vault, and performs the appropriate rotation for each secret that requires it. Failures in this step are caused by secrets that require rotation and can't be rotated manually. To manually rotate these secrets a human will need to run the `synchronize` command manually and specify the manifest that contains the secret requiring rotation.
+
+
+## Creating new secrets
+New secrets can be created by adding new entries to the manifest file, then running synchronize. The tool will then create the new secret and add it to the secret store.
+
 ## Secret Types
 Each secret in the manifest has a specified type, and these types determine what parameters are required, and what ends up in the secret store. Some secret types require human interaction to generate and/or rotate. For such secrets the tool will prompt the user or fail if in a build context. Some secret types also produce multiple values. These values are all stored with additional suffixes in the secret store. Each type is documented below.
 
 ### Azure Storage
 
+#### Storage Key
+Only use this secret type if the connection string is required in multiple vaults. The storage key should be in one of them, and each vault can have a connection string secret that references the key.
+```yaml
+type: azure-storage-key
+parameters:
+  account: storage account name
+  subscription: Azure subscription id
+```
+
 #### Connection String
 ```yaml
 type: azure-storage-connection-string
 parameters:
-  subscription: Azure subscription id
   account: storage account name
+  # one of the following is required
+  storageKeySecret: SecretReference to an azure-storage-key
```

Comment:
Sorry if I'm missing this, where does one learn how to express a SecretReference?

Reply:
There is an example in the manifest at the top.